docs/language_guide.md

```diff
@@ -0,0 +1,619 @@
+# Language Guide
```

Comment:
perhaps it'd be good to have both tutorial style documentation (like this) and language reference documentation.

i'd probably be able to understand more of what was going on here if i was able to read this at the same time as reading some language reference